tf2_sensor_msgs/include/tf2_sensor_msgs/tf2_sensor_msgs.h

```diff
@@ -86,6 +88,151 @@ void fromMsg(const sensor_msgs::PointCloud2 &msg, sensor_msgs::PointCloud2 &out)
   out = msg;
 }
 
+/**********/
+/** IMU  **/
+/**********/
+
+/**
+* method to extract timestamp from object
+*/
+template <>
+inline
+const ros::Time& getTimestamp(const sensor_msgs::Imu& p) {return p.header.stamp;}
+
+/**
+* method to extract frame id from object
+*/
+template <>
+inline
+const std::string& getFrameId(const sensor_msgs::Imu &p) {return p.header.frame_id;}
+
+
+/**
+* Transforms a covariance array from one frame to another
+*/
+inline
+void transformCovariance(const boost::array<double, 9>& in, boost::array<double, 9>& out, Eigen::Quaternion<double> r){
+
+  Eigen::Map<const Eigen::Matrix<double, 3, 3, Eigen::RowMajor> > cov_in(in.data());
+  Eigen::Map<Eigen::Matrix<double, 3, 3, Eigen::RowMajor> > cov_out(out.data());
+  cov_out = r * cov_in * r.inverse();
+
+}
+
+/**
+* Transforms sensor_msgs::Imu data from one frame to another
+*/
+template <>
+inline
+void doTransform(const sensor_msgs::Imu &imu_in, sensor_msgs::Imu &imu_out, const geometry_msgs::TransformStamped& t_in)
+{
+
+  imu_out.header = t_in.header;
+
+  // Discard translation, only use orientation for IMU transform
+  Eigen::Quaternion<double> r(
+      t_in.transform.rotation.w, t_in.transform.rotation.x, t_in.transform.rotation.y, t_in.transform.rotation.z);
+  Eigen::Transform<double,3,Eigen::Affine> t(r);
+
+  Eigen::Vector3d vel = t * Eigen::Vector3d(
+      imu_in.angular_velocity.x, imu_in.angular_velocity.y, imu_in.angular_velocity.z);
+
+  imu_out.angular_velocity.x = vel.x();
+  imu_out.angular_velocity.y = vel.y();
+  imu_out.angular_velocity.z = vel.z();
+
+  transformCovariance(imu_in.angular_velocity_covariance, imu_out.angular_velocity_covariance, r);
+
+  Eigen::Vector3d accel = t * Eigen::Vector3d(
+      imu_in.linear_acceleration.x, imu_in.linear_acceleration.y, imu_in.linear_acceleration.z);
+
+
+  imu_out.linear_acceleration.x = accel.x();
+  imu_out.linear_acceleration.y = accel.y();
+  imu_out.linear_acceleration.z = accel.z();
+
+  transformCovariance(imu_in.linear_acceleration_covariance, imu_out.linear_acceleration_covariance, r);
+
+  Eigen::Quaternion<double> orientation = r * Eigen::Quaternion<double>(
```

Comment:
Shouldn't orientation be `imu_in.orientation*r.inverse()`, instead of `r*imu_in.orienation*r.inverse()` ?
I tried this code with an IMU mounted tilted on an object and `imu_in.orientation*r.inverse()` works for me (unless I'm misunderstanding what it should do. I had assumed when having an IMU mounted on an object with transform only being a rotation around y-axis, when the object it's mounted on is level pointing east, the output of transform should be unit quaternion?)